user_model.ml

```diff
@@ -56,9 +56,12 @@ let cookie_to_json (cookie : cookie) : Yojson.Basic.t =
     ]
 
 let string_or_none field = function
-  | None -> Ok None
+  | None | Some `Null -> Ok None
```

Comment:
this is the fix needed. all the other changes are just for more informative error messages.